lib/web/users.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// requestUser is used to pull out data from JSON requests.
+type requestUser struct {
+	Name  string   `json:"name"`
+	Roles []string `json:"roles"`
+}
+
+// responseUserPasswordToken is used to send back data
+// about created user and password setup token.
+type responseUserPasswordToken struct {
+	User  *services.UserV2               `json:"user"`
```

Comment:
UI uses its own set of objects, please add one for the users (/web/ui). This way you control which fields are consumed by UI to avoid breaking changes.

Reply:
revised